ipaclient/csrgen.py

```diff
@@ -407,11 +407,11 @@ def __init__(self, key_filename, password_filename):
         self.password_filename = password_filename
 
     def key(self):
-        with open(self.key_filename, 'r') as key_file:
+        with open(self.key_filename, 'rb') as key_file:
             key_bytes = key_file.read()
         password = None
         if self.password_filename is not None:
-            with open(self.password_filename, 'r') as password_file:
+            with open(self.password_filename, 'rb') as password_file:
```

Comment:
Do we need to read passwords as bytes?

Reply:
Hi @stlaz 
thank you for your comment. Yes, according to https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#pem the method load_pem_private_key expects bytes for key_bytes and password.

Reply:
Ah, thanks. I'll perform a sanity test and will ACK this afterwards :)